packages/hash/api/src/graph/workspace-types.ts

```diff
@@ -61,26 +59,29 @@ export const ensureWorkspaceTypesExist = async (params: {
   );
 
   // Next, let's ensure all workspace property types have been created
-  await Promise.all(
-    workspacePropertyTypes.map(async (schema) => {
-      const { $id: versionedUri } = schema;
-      const existingPropertyType = await PropertyTypeModel.get(graphApi, {
-        versionedUri,
-      }).catch((error: AxiosError) =>
-        error.response?.status === 404 ? null : Promise.reject(error),
-      );
+  // This is done sequentially as property types might reference other property
+  // types
+  /**
+   * @todo Use transactional primitive/bulk insert to be able to do this in parallel
+   *   see the following task:
+   *   https://app.asana.com/0/1201095311341924/1202573572594586/f
+   */
+  for (const schema of workspacePropertyTypes) {
+    const { $id: versionedUri } = schema;
+    const existingPropertyType = await PropertyTypeModel.get(graphApi, {
```

Comment:
Not asking for a change, but I wonder if we'd prefer to try and optimistically insert and catch the correct error code. I'm not sure we have decided on an appropriate response though, we might want to look into that.

(A quick google suggests we'd need to think about a 2XX code, vs something like a 409)

Reply:
That would make this a little simpler. We don't really have any issues since we don't do "upserts".
I think we should look into that in a follow-up [asana task](https://app.asana.com/0/1202805690238892/1202898350934175/f)

Reply:
Yeah I'm just thinking about more tooling and blocks. We don't want them to **have** to make a GET first if they're trying to create. 

I think if we carefully and comprehensively define our return statuses it should be possible to have simple solutions for stuff like
- Insert and fail if it already exists
- Insert if it doesn't exist already
- Update, or create if it doesn't exist
etc.

Reply:
Added this comment to the description of [this task](https://app.asana.com/0/1201095311341924/1202574350052904/f) as an example usage